src/render/PanelHeaderProvider.js

```diff
@@ -191,4 +187,12 @@ function getTemplateDocumentation(element, elementTemplates) {
   const template = getTemplate(element, elementTemplates);
 
   return template && template.documentationRef;
+}
+
+function getModelerTemplateIcon(element) {
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const config = useService('config.elementTemplateIconRenderer', false);
+  const { iconProperty = 'zeebe:modelerTemplateIcon' } = config || {};
```

Comment:
As this component is generic, we don't want to assume `zeebe:modelerTemplateIcon` as a default here. If there is no configured `iconProperty`, we simply want to fall back to the original icon.

Reply:
I made the judgment here. Because the default value on the icon template is 'zeebe:modelerTemplateIcon'.And I won't set the icon template config for normal use.
<img width="794" alt="image" src="https://github.com/bpmn-io/bpmn-js-properties-panel/assets/62740686/7284668c-6d55-4240-9e36-a623b8d4206a">

Reply:
Are you sure you want to remove the default values???

Reply:
@philippfromme On the other hand it wants to play nicely with our other utilities (i.e. the icons renderer). There we assume a default.

Hence I think it is safe to keep it hard coded per default _or_ get the icon from the icon renderer (if it exists).

Reply:
Okay, makes sense. We can go ahead with it then. 👍🏻